cmd/grype/cli/commands/root.go

```diff
@@ -155,6 +155,15 @@ func runGrype(app clio.Application, opts *options.Grype, userInput string) (errs
 		opts.Ignore = append(opts.Ignore, ignoreFixedMatches...)
 	}
 
+	for _, ignoreState := range opts.IgnoreStates {
+		switch grypeDb.FixState(ignoreState) {
+		case grypeDb.UnknownFixState, grypeDb.FixedState, grypeDb.NotFixedState, grypeDb.WontFixState:
+			opts.Ignore = append(opts.Ignore, match.IgnoreRule{FixState: ignoreState})
+		default:
+			log.Warnf("ignoring unknown fix state %s for --ignore-states", ignoreState)
```

Comment:
I think this should return an error instead of logging a warning; the CLI should fail and explain why if the user invokes it with invalid config.

Reply:
I've addressed this on the latest commit, and I've also moved this (and the other ignore state handling) up so they happen prior to downloading the DB and running the scan, to enable a fast-feedback loop if a user supplies an invalid fix state (rather than waiting for the DB update & scan to run before they are made aware of the their typo :smile:)